app/views/govuk_publishing_components/components/_summary_list.html.erb

```diff
@@ -1,23 +1,47 @@
 <%
+  local_assigns[:heading_level] ||= 3
+  heading_size = 'm' unless ['s', 'm', 'l', 'xl'].include?(heading_size)
+  shared_helper = GovukPublishingComponents::Presenters::SharedHelper.new(local_assigns)
+
   id ||= nil
   title ||= nil
   borderless ||= false
   edit ||= {}
+  delete ||= {}
   items ||= []
   block ||= yield
 %>
 <% if title || items.any? %>
   <%= tag.div class: "gem-c-summary-list #{"govuk-summary-list--no-border" if borderless}", id: id do %>
     <% if title %>
-      <%= tag.h3 title, class: "govuk-heading-m" %>
-      <% if edit.any? %>
-        <%= tag.ul class: "govuk-summary-list__actions-list" do %>
-          <%= tag.li class: "govuk-summary-list__actions-list-item" do %>
-            <%= link_to edit.fetch(:href),
-                      class: "govuk-link gem-c-summary-list__edit-section-link",
-                      title: "#{edit.fetch(:link_text, "Change")} #{title}",
-                      data: edit.fetch(:data_attributes, {}) do %>
-               <%= edit.fetch(:link_text, "Change") %> <%= tag.span title, class: "govuk-visually-hidden" %>
+      <%= content_tag(
+            shared_helper.get_heading_level,
+            title,
+            class: "govuk-heading-#{heading_size} gem-c-summary-list__group-title"
+          )
+      %>
+      <% if edit.any? || delete.any? %>
+        <%= tag.ul class: "govuk-summary-list__actions-list gem-c-summary-list__group-actions-list" do %>
+          <%- if edit.any? %>
+            <% edit_section_link_text = edit[:link_text] || "Change" %>
```

Comment:
Does this text (and "Delete" on line 37) need to go into the `en.yml` file?

Reply:
Great improvement suggestion, have moved the hardcoded text in locales. We should raise issues for components that still have hardcoded text in order to align them.

Reply:
This is actually the thing I usually forget about when working on components, but I suddenly remembered it's one of our core principles! Good idea to raise issues, I've added a note to the component audit card in the backlog: https://trello.com/c/OvXd2E84/103-component-audit